### PR TITLE
Remove global error tracking always wait

### PR DESCRIPTION
**Pull request by wudi**
Fix issue: https://github.com/micro/go-micro/issues/1750  not wait while occur error on Recv

Remove global error tracking and always wait coroutines done.

### CODE REVIEW

**Comment by sadwxqezc** [comment]
LGTM

**Comment by vtolstov** [comment]
looks good, but needs rebase

**Comment by wudi** [comment]
> looks good, but needs rebase

done

**Comment by vtolstov** [comment]
@asim we have failed test because examples not fixed by @ben-toogood 
i think that we can merge this.
```
build github.com/micro/examples/sharding: cannot load github.com/micro/go-micro/v2/client/selector: module github.com/micro/go-micro/v2@latest found (v2.9.0, replaced by github.com/wudi/go-micro/v2@v2.9.1-0.20200702154856-2846e957bbab), but does not contain package github.com/micro/go-micro/v2/client/selector
```

File: server/rpc_server.go [hunk 1]
```diff
@@ -303,14 +293,12 @@ func (s *rpcServer) ServeConn(sock transport.Socket) {
 			// try get a new codec
 			if cf, err = s.newCodec(ct); err != nil {
 				// no codec found so send back an error
-				if err := sock.Send(&transport.Message{
+				sock.Send(&transport.Message{
```
Comment (vtolstov) [inline server/rpc_server.go:296]: why error not checked?

Comment (wudi) [reply server/rpc_server.go:296]: > why error not checked?

Error check previous just set `gerr` for global error tracking, unnecessary, now.

Comment (vtolstov) [reply server/rpc_server.go:296]: yes, but for now - if sock.Send returns error why we not log it and not use it ?

Comment (vtolstov) [reply server/rpc_server.go:296]: if it not needed for client i think that we need to log it via logger.Errorf

**Pull request merged**